src/clarity-angular/datagrid/render/column-resizer.ts

```diff
@@ -0,0 +1,190 @@
+                private organizer: DatagridRenderOrganizer,
+                private domAdapter: DomAdapter) {
+
+        this.columnEl = el.nativeElement;
+    }
+
+    columnEl: Element;
+    columnRectWidth: number;
+    columnResizeBy: number = 0;
+
+    draggerEl: Element;
+    draggerElWidth: number;
+    draggerTracker: Element;
+
+    pageStartPositionX: number;
+    dragDistancePositionX: number; //relative to pageStartPosition
+
+    @Output("clrDgColumnResize") resizeEmitter: EventEmitter<any> = new EventEmitter();
+
+    dragWithinMinWidth: boolean = false;
+
+    private _columnMinWidth: number;
+
+    get columnMinWidth(): number {
+        return this._columnMinWidth;
+    }
+
+    set columnMinWidth(value: number) {
+        this._columnMinWidth = value || 96;
+    }
+
+    private columnResizeEvents: Function[];
+
+    ngAfterViewInit() {
+        this.columnResizeEvents = this.addDragListener();
+    }
+
+    ngOnDestroy() {
+        this.columnResizeEvents.map(event => event());
+    }
+
+
+    addDragListener(): Function[] {
+
+        this.draggerEl = this.columnEl.querySelector("button.separator-dragger");
+        this.draggerTracker = this.columnEl.querySelector(".separator-tracker");
+
+        this.draggerElWidth = this.domAdapter.clientRectWidth(this.draggerEl);
+
+        this.columnMinWidth = this.domAdapter.minWidth(this.columnEl);
+
+        return this.listenDrag();
+    }
+
+    listenDrag(): Function[] {
+
+        let mouseDownListener: any;
+        let mouseMoveListener: any;
+        let mouseUpListener: any;
+
+        let touchStartListener: any;
+        let touchMoveListener: any;
+        let touchEndListener: any;
+
+        mouseDownListener = this.renderer.listen(this.draggerEl, "mousedown", () => {
+            this.onDragStart();
+
+            mouseMoveListener = this.renderer.listenGlobal("document", "mousemove", (mouseMoveEvent: any) => {
+                this.onDragMove(mouseMoveEvent);
+            });
+
+            mouseUpListener = this.renderer.listenGlobal("document", "mouseup", () => {
+
+                //Unsubscribing from mouseMoveListener
+                mouseMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                mouseUpListener();
+            });
+        });
+
+        touchStartListener = this.renderer.listen(this.draggerEl, "touchstart", () => {
+            this.onDragStart();
+
+            touchMoveListener = this.renderer.listenGlobal("document", "touchmove", (touchMoveEvent: any) => {
+                this.onDragMove(touchMoveEvent);
+            });
+
+            touchEndListener = this.renderer.listenGlobal("document", "touchend", () => {
+
+                //Unsubscribing from mouseMoveListener
+                touchMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                touchEndListener();
+            });
+        });
```

Comment:
Please make a method abstracting the "start, move, stop" pattern so that we don't have the exact same code twice, one for mouse and one for touch.